lib/do_image/do_tag.js

```diff
@@ -0,0 +1,130 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2020 Joyent, Inc.
+ *
+ * `triton image tag ...`
+ */
+
+var vasync = require('vasync');
+
+var common = require('../common');
+var errors = require('../errors');
+var mat = require('../metadataandtags');
+
+// ---- the command
+
+function do_tag(subcmd, opts, args, cb) {
+    if (opts.help) {
+        this.do_help('help', {}, [subcmd], cb);
+        return;
+    } else if (args.length < 1) {
+        cb(new errors.UsageError('incorrect number of args'));
+        return;
+    }
+
+    var log = this.top.log;
+    var tritonapi = this.top.tritonapi;
+
+    vasync.pipeline({arg: {cli: this.top}, funcs: [
+        common.cliSetupTritonApi,
+
+        function gatherTags(ctx, next) {
+            mat.tagsFromSetArgs(opts, args.slice(1), log, function (err, tags) {
+                if (err) {
+                    next(err);
+                    return;
+                }
+                log.trace({tags: tags || '<none>'},
+                    'tags loaded from opts and args');
+                ctx.tags = tags;
+                next();
+            });
+        },
+        function updateImageTags(ctx, next) {
+            log.trace({dryRun: opts.dry_run, tags: ctx.tags},
+                'image tag');
+
+            if (opts.dry_run) {
+                next();
+                return;
+            }
+
+            tritonapi.updateImage({
+                image: args[0],
+                fields: {
+                    tags: ctx.tags
+                }
+            }, function (err, img) {
```

Comment:
Is it intentional that it would remove all existing tags, e.g.:
```
$ ./bin/triton image get 506cdbc2  | json tags
{
  "foo": "bar"
}

$ ./bin/triton image tag 506cdbc2 foo=bar bar=baz
Updated image 506cdbc2 with tags {"foo":"bar","bar":"baz"}
$ ./bin/triton image get 506cdbc2  | json tags
{
  "foo": "bar",
  "bar": "baz"
}

# This will lose bar=baz tag:
$ ./bin/triton image tag 506cdbc2 foo=bar
Updated image 506cdbc2 with tags {"foo":"bar"}
$ ./bin/triton image get 506cdbc2  | json tags
{
  "foo": "bar"
}
```

Reply:
Yeah. I've updated the subcommand information to make clear that it will override any existing value. I'd say that allowing to add new tags and keep the existing ones makes it difficult to provide a way to get rid of those existing tags w/o adding the whole set of add/remove/reset tags collection of subcommands. Anyway, happy to modify to always keep existing tags too and allow tag removal only through `image update`